daemon/inspect.go

```diff
@@ -112,10 +112,9 @@ func (daemon *Daemon) getInspectData(container *container.Container, size bool)
 	// container will use daemon defaults even if daemon changes them
 	if hostConfig.LogConfig.Type == "" {
 		hostConfig.LogConfig.Type = daemon.defaultLogConfig.Type
-	}
-
-	if len(hostConfig.LogConfig.Config) == 0 {
```

Comment:
Shouldn't we keep it like this but add a ` && hostConfig.LogConfig.Type == daemon.defaultLogConfig.Type`